httplug/users.rst

```diff
@@ -1,30 +1,44 @@
 HTTPlug for library users
 =========================
 
-If you use a library that depends on HTTPlug (say, ``some/awesome-library``),
-you need to include an HTTPlug client in your project before you can include
-``awesome-library``.
+This page explains how to set up a library that depends on HTTPlug.
 
-First you need to choose library to send HTTP messages and that also provide the virtual package
-`php-http/client-implementation`_. In the example
-below we are using cURL.
+TL&DR
+-----
 
-.. note::
-
-    Read about the clients and adapters provided by the PHP-HTTP organization at :doc:`this page </clients>`.
+For the impatient: Require the following packages before requiring the library
+you plan to use:
 
 .. code-block:: bash
 
-    $ composer require php-http/curl-client
+    composer require php-http/curl-client guzzlehttp/psr7 php-http/message
+
+Background
+----------
+
+If a library depends on HTTPlug, it requires the virtual package
+`php-http/client-implementation`_. A virtual package is used to declare that
+the library needs *an* implementation of the HTTPlug interfaces, but does not
+care which implementation specifically.
+
+The client-implementation is provided by all HTTPlug compliant clients and
+adapters. Read about the clients and adapters provided by the PHP-HTTP
+organization at :doc:`this page </clients>`.
```

Comment:
Is this paragraph needed? We reference the `clients` doc page below. It is also an interruption between paragraph 1 and 3 which seams more natural together. 

I suggest removing this paragraph.